lib/user.rb

```diff
@@ -0,0 +1,24 @@
+class User
+
+attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+    @jokes
+  end
+
+  def tell(recipient, joke)
+    recipient.jokes << joke
+  end
+
+  def joke_by_id(id)
+    @jokes.find_all do |joke|
+      id == joke.id
+    end
```

Comment:
really great here -- in this case, since we're only looking for one joke by its id, we can just use `find` instead of `find_all` and then it'll stop looking through `@jokes` as soon as it finds a match; `find` can use the same boolean expression as you already have